compiler/damlc/tests/daml-test-files/SemanticsEvalOrderSince1_11.daml

```diff
@@ -0,0 +1,41 @@
+
+-- | Test operational semantics for 1.11 features.
+--
+-- @SINCE-LF 1.11
+module SemanticsEvalOrderSince1_11 where
+
+-- @ERROR EvUpdExercWithoutActorsErr OK
```

Comment:
the order seems a bit surprising. We enforce an order at the parser level where `observer` must come before `controller` but then we evaluate it the other way around. Is there a particular reason for this choice? It seems like evaluating observers first might be more intuitive and I don’t see why it should be harder from an implementation pov.

Reply:
Not sure there's a good reason, but it's the semantics in LF 1.11.

Here's a bad justification: From the point of view of the spec, in an "exercise without actors" (which this is) we need to determine the exercise actor via the controller field, before we get an "exercise with actors", which is where all the semantics for an exercise (including evaluating observers) is defined.

I think "exercise with actors" is an outdated notion and should probably be removed from the spec, at some point? (cc @remyhaemmerle-da)

The fact that observers have to come before controllers in the syntax is a bit of an accident. IIRC Martin preferred it to be flexible, or to have it fixed the other way, but Nick ran into issues trying to implement it that way in the GHC parser.